bottomless/src/replicator.rs

```diff
@@ -693,23 +682,46 @@ impl Replicator {
         tracing::debug!("Rolled back to {}", last_valid_frame);
     }
 
+    // Opens a raw libSQL connection that doesn't checkpoint.
+    // Useful for reading metadata from the database file.
+    fn open_db(&self) -> Result<libsql_sys::Connection<libsql_sys::wal::Sqlite3Wal>> {
+        use libsql_sys::connection::OpenFlags;
+        use libsql_sys::wal::Sqlite3WalManager;
+        let flags = OpenFlags::SQLITE_OPEN_READ_WRITE
+            | OpenFlags::SQLITE_OPEN_CREATE
+            | OpenFlags::SQLITE_OPEN_URI
+            | OpenFlags::SQLITE_OPEN_NO_MUTEX;
+
+        let conn = libsql_sys::Connection::open(
+            &self.db_path,
+            flags,
+            Sqlite3WalManager::new(),
+            u32::MAX, // no checkpointing
```

Comment:
Wasn't `0` a value for turning checkpoints off?

Reply:
Right. Technically it's just as good, because we're not adding 4M pages with this short-lived connection, but it's more correct to use 0. I'll make it a constant in `libsql-sys/connection` and use it